tests/examples/setuptools-with-imports-in-setup-py/dependencies.yaml

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+files:
```

Comment:
> *Rather than creating a new examples directory for tests, can you reuse the existing templates?*

@vyasr How strongly do you feel about this?

I explored that briefly, but found that I'd need to add significant new complexity to the templates and corresponding functions for using them in testing.

And I really feel the tests I've written here are stronger as a result of NOT relying on those templates.

For example, they:

* don't rely on CUDA 11 and 12 being used (they use made-up versions 25 and 85)
  - *avoids any "if CUDA 12" types of logic making it into these codepaths inadvertently*
* use different keys in `files:` and `dependencies:` in `dependencies.yaml` than the templates
  - *avoids accidentally relying on *conventions* about how certain sections items there are named*
* test the `setuptools` behavior without caring about CUDA-suffixing of dependencies
   - *avoids that logic accidentally being limited in any way to just CUDA-suffixed dependencies*
* rely on a `dependencies.yaml` that does not have anything in `files:` for updating `[build-system]` in `pyproject.toml`
  - *would catch any changes that make the existence of such a block a hard requirement*

If you read this and don't find those reasons convincing, then I'll update this to use the existing templates and extend them to support the new tests I've added in this PR.

Reply:
I should rephrase my request. I didn't mean that you have to use the exact templates that are already there. I agree that the dependencies.yaml file you have here is much simpler and easier to work with for this test. The only reason that there is one and only one example of each file type in the templates right now is because nobody has updated those since I wrote them. I think it's perfectly fine if you want to add a new `dependencies_simple.yaml` file, for instance.

OTOH, I think you can reuse the pyproject.toml template almost verbatim. I don't see anything in there that would convolute your testing, do you? More importantly, if you added a setup.py template you'd be able to do things like add one test with `setup_requires` and one without, which right now would require you to copy-paste the entire example. Essentially my suggestion isn't to use the templates that exist, it's to use ones that are helpful and add others if needed, but to leverage the templating framework to make it easier to improve test coverage.

Reply:
> *if you added a setup.py template you'd be able to do things like add one test with setup_requires and one without, which right now would require you to copy-paste the entire example.*

Well for one, the templating would need a way to express **not generating that file at all** (not just which values to template through it), since there are code paths being added in this PR whose behavior depend on whether or not a file like that even exists.

Totally manageable to do, but I'm not convinced that that's actually simpler than what this PR already has.

This PR already has PRs testing `setup.py` with and without `setup_requires`, and all it took was this:

```python
with open(package_dir / "setup.py", "w") as f:
    f.write("from setuptools import setup\nsetup(setup_requires=['Cython'])\n")
```

>  *...reuse the pyproject.toml template almost verbatim. I don't see anything in there that would convolute your testing, do you?*

You're totally right, the `pyproject.toml` looks like it makes a very minimal set of assumptions, and should work as-is for the tests in this PR.

> *Essentially my suggestion isn't to use the templates that exist, it's to use ones that are helpful and add others if needed, but to leverage the templating framework to make it easier to improve test coverage.*

Alright I'll do that.

Reply:
> the templating would need a way to express not generating that file at all

Looking at this more closely, I understand now that this was wrong. I see that the tests are setting up an empty directory, and then copying files into that directory 1 at a time based on calls to `generate_from_template()`

https://github.com/rapidsai/rapids-build-backend/blob/d2202dd19b08f9f99ea047212ad74720844eb13f/tests/test_packages.py#L69-L73

Not sure why I thought yesterday that all those files were being copied into every directory created by this, sorry 😬

Reply:
I just pushed commits switching the tests to using the templating. It was pretty straightforward with the constraint "use the exact templates that are already thee" removed.